src/core/Akka.Cluster/ClusterDaemon.cs

```diff
@@ -1015,7 +1015,7 @@ public ClusterCoreDaemon(IActorRef publisher)
             _cluster = Cluster.Get(Context.System);
             _publisher = publisher;
             SelfUniqueAddress = _cluster.SelfUniqueAddress;
-            _vclockNode = new VectorClock.Node(VclockName(SelfUniqueAddress));
+            _vclockNode = VectorClock.Node.Create(VclockName(SelfUniqueAddress));
```

Comment:
THIS IS THE FIX FOR #2584 - as it turns out, the constructor we were using earlier didn't compute a hash of the node - thus all of the comparisons would fail later when it came time to prune vector clocks for removed / downed nodes.